designer/server/src/main/scala/pl/touk/nussknacker/ui/services/NuDesignerApiAvailableToExpose.scala

```diff
@@ -0,0 +1,73 @@
+package pl.touk.nussknacker.ui.services
+
+import org.reflections.Reflections
+import org.reflections.util.ConfigurationBuilder
+import pl.touk.nussknacker.ui.api.BaseEndpointDefinitions
+import pl.touk.nussknacker.ui.security.api.AuthCredentials
+import sttp.apispec.openapi.circe.yaml.RichOpenAPI
+import sttp.tapir.EndpointInput.Auth
+import sttp.tapir.docs.openapi.{OpenAPIDocsInterpreter, OpenAPIDocsOptions}
+import sttp.tapir.{Endpoint, auth}
+
+import scala.jdk.CollectionConverters._
+import java.lang.reflect.{Method, Modifier}
+import scala.util.Try
+
+object NuDesignerApiAvailableToExpose {
+
+  val name = "Nussknacker Designer API"
+
+  val openAPIDocsOptions: OpenAPIDocsOptions = OpenAPIDocsOptions.default
+    .copy(markOptionsAsNullable = true)
+
+  def generateOpenApiYaml: String = {
+    val endpoints = findApiEndpointsClasses().flatMap(findEndpointsInClass)
+    val docs = OpenAPIDocsInterpreter(openAPIDocsOptions).toOpenAPI(
+      es = endpoints,
+      title = NuDesignerApiAvailableToExpose.name,
+      version = ""
+    )
+
+    docs.toYaml
+  }
+
+  private def findApiEndpointsClasses() = {
+    val baseEndpointDefinitionsClass = classOf[BaseEndpointDefinitions]
+    val reflections = new Reflections(new ConfigurationBuilder().forPackages(baseEndpointDefinitionsClass.getPackageName))
```

Comment:
Can't we just have a list of definitions to be exposed in the same place as they exposed (`AppApiHttpService`) instead of using the "magic" of reflection? What if we want to bind two interfaces and expose one part of the API on one port and another on the second?

Reply:
I think that it is our near future. We want to have an external API for things like batch invocations. But let's start with this approach and see what's happened then.